public/buttonComponent.js

```diff
@@ -0,0 +1,45 @@
+export class ButtonComponent extends HTMLElement {
+    constructor() {
+        super();
+        this.selectButtons =[];
```

Comment:
```suggestion
        this.selectButtons=[];
```

コードを見る感じ`=`の廻りにスペース入れない主義っぽいのでここもなくていいと思います。
（本来のコーディング規約的な話はちょっとあれかもです）